_posts/2020-10-26-dynamically-changeable-logging.md

```diff
@@ -0,0 +1,150 @@
+```
+
+Strimzi Cluster Operator, Topic Operator, Entity Operator and Strimzi Kafka Bridge use automatic reloading of logging configuration.
+The only way to check the logging configuration of these components is by looking into the pod for the file on the path `/opt/strimzi/custom-config/log4j2.properties`.
+It should contain the configuration and the `monitorInterval` property.
+Note that reconciliation time and `monitorInterval` value affect the time logging configuration to be reloaded.
+
+## Implementation details
+
+To implement dynamically changeable logging levels in Strimzi Cluster Operator, Topic Operator, Entity Operator and Strimzi Kafka Bridge we use the`log4j2` [automatic reconfiguration](https://logging.apache.org/log4j/log4j-2.1/manual/configuration.html#AutomaticReconfiguration) feature.
+
+Automatic reloading works in these steps:
+1. Logging configuration is changed in the custom resource
+2. During reconciliation the `log4j2.properties`  file is remounted. It contains the configuration from the custom resource.
+3. `log4j2` polls changes in the `log4j2.properties` file in an interval of `monitorInterval` seconds
+
+For Kafka broker, Kafka Connect and Kafka MirrorMaker2 we use features already implemented from Kafka.
+Logging configuration in Kafka Brokers is changed using `AdminClient`.
+For more information, see [KIP-412](https://cwiki.apache.org/confluence/display/KAFKA/KIP-412%3A+Extend+Admin+API+to+support+dynamic+application+log+levels)
+
+Kafka Connect and Kafka MirrorMaker2 logging configuration is changed using the REST API.
+For more information, see [KIP-495](https://cwiki.apache.org/confluence/display/KAFKA/KIP-495%3A+Dynamically+Adjust+Log+Levels+in+Connect)
+
+## Further steps
+
+For now, there is no way to set logging configuration for Kafka MirrorMaker.
+There is a [KIP-649](https://cwiki.apache.org/confluence/display/KAFKA/KIP-649%3A+Dynamic+Client+Configuration) in progress.
+However, Kafka MirrorMaker support will eventually be dropped, as it has been replaced by MirrorMaker 2.0.. 
+
+
+An upgrade to `log4j` has been proposed: [KIP-653](https://cwiki.apache.org/confluence/display/KAFKA/KIP-653%3A+Upgrade+log4j+to+log4j2) arised.
+For now, it is not clear how this upgrade will change dynamic changes to logging levels.
+
+## Conclusion
+
+Whether Strimzi is running in development or production environments, it is useful to be able to set an appropriate logging level.
+Sometimes a Kafka cluster or operator is driven into an edge case.
+With a less informative logging level it can be difficult to investigate what happened and how this situation can be fixed.
+Changing the logging level causes a rolling update of the pod, which can put the cluster in a regular state.
+This complicates debugging the cause of such an event.
+Furthermore, any reason to roll Kafka pods lowers the availability of Kafka cluster, which is the main reason why we decided to implement dynamic log levels in Strimzi 0.20.0.
```

Comment:
I like that you repeat the problem. But maybe you can add few more sentences around how it is now solved. To make it a bit more stand out.